packages/yew/src/renderer.rs

```diff
@@ -0,0 +1,94 @@
+use std::cell::Cell;
+use std::panic::PanicInfo;
+use std::rc::Rc;
+
+use web_sys::Element;
+
+use crate::app_handle::AppHandle;
+use crate::html::IntoComponent;
+
+thread_local! {
+    static PANIC_HOOK_IS_SET: Cell<bool> = Cell::new(false);
+}
+
+/// Set a custom panic hook.
+/// Unless a panic hook is set through this function, Yew will
+/// overwrite any existing panic hook when one of the `start_app*` functions are called.
+#[cfg_attr(documenting, doc(cfg(feature = "render")))]
+pub fn set_custom_panic_hook(hook: Box<dyn Fn(&PanicInfo<'_>) + Sync + Send + 'static>) {
+    std::panic::set_hook(hook);
+    PANIC_HOOK_IS_SET.with(|hook_is_set| hook_is_set.set(true));
+}
+
+fn set_default_panic_hook() {
+    if !PANIC_HOOK_IS_SET.with(|hook_is_set| hook_is_set.replace(true)) {
+        std::panic::set_hook(Box::new(console_error_panic_hook::hook));
+    }
+}
+
+/// The Yew Renderer.
+///
+/// This is the main entry point of a Yew application.
+#[derive(Debug)]
+#[cfg_attr(documenting, doc(cfg(feature = "render")))]
+#[must_use = "Renderer does nothing unless render() is called."]
+pub struct Renderer<ICOMP>
+where
+    ICOMP: IntoComponent + 'static,
+{
+    root: Element,
+    props: ICOMP::Properties,
+}
+
+impl<ICOMP> Default for Renderer<ICOMP>
+where
+    ICOMP: IntoComponent + 'static,
+    ICOMP::Properties: Default,
+{
+    fn default() -> Self {
+        Self::with_props(Default::default())
+    }
+}
+
+impl<ICOMP> Renderer<ICOMP>
```

Comment:
I'm thinking maybe this could be named `YewDOM` same way as react. this way we can add more than just a render function here.
imo would look a lot more neat:
```rust
yew::YewDOM::<App>::new().render();
```

Reply:
There have been a naming discussion in a pervious pull request (https://github.com/yewstack/yew/pull/2335#discussion_r780347806).

The `yew::Renderer` naming is consistent with the `yew::ServerRenderer`.

I think unless `ServerRenderer` is renamed to something like `YewServerDom`, then I guess `yew::Renderer` naming is better.
(Also, I guess the name needs to be `YewDom` or clippy is going to complain.)

Reply:
Yeah then `YewDOMServer` would be nice
For clippy you can just add `allow(clippy::rule)` as having `om` in lover case does not make sense as its a abbreviation for Document Object Model

Reply:
> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn. In snake_case, acronyms and contractions are lower-cased: is_xid_start.

https://rust-lang.github.io/api-guidelines/naming.html

Do we need to go this far (disabling clippy and breaking Rust naming conventions) to have the same name as React?